src/storage/datacore/restapi/mode/statusmonitor.pm

```diff
@@ -0,0 +1,151 @@
+
+sub new {
+    my ($class, %options) = @_;
+
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+    $options{options}->add_options(arguments => {
+        'filter-caption:s' => { name => 'filter_caption' } });
+    $self->{cache} = centreon::plugins::statefile->new(%options);
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+    $self->{cache}->check_options(option_results => $self->{option_results});
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 1, message_multiple => 'All memory usages are ok', skipped_code => { -10 => 1 } }
+    ];
+
+    $self->{maps_counters}->{global} = [
+        { label              => 'state',
+            type             => 2,
+            warning_default  => '%{state} =~ /Warning/i',
+            critical_default => '%{state} =~ /Critical/i',
+            set              => {
+                key_values                     => [ { name => 'state' }, { name => 'extendedcaption' }, { name => 'messagetext' } ],
+                closure_custom_output          => $self->can('custom_status_output'),
+                closure_custom_perfdata        => sub {return 0;},
+                closure_custom_threshold_check => \&catalog_status_threshold_ng
+            }
+        }
+    ];
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+    my $monitor_data = $self->request_monitors(%options);
+    my $i = 0;
+    for my $object (@$monitor_data) {
+
+        if (!centreon::plugins::misc::is_empty($self->{option_results}->{filter_caption})
+            and $object->{ExtendedCaption} !~ $self->{option_results}->{filter_caption}) {
+            next;
+        }
+        $self->{global}->{$i} = {
+            state           => $monitor_state{$object->{State}},
+            messagetext     => $object->{MessageText},
+            extendedcaption => $object->{ExtendedCaption}
+        };
+        $i++;
+    }
+    # for now if no data is given to the counter, output is OK: with status ok, instead of unknown.
+    # We manage this case in each plugin for now.
+    if ($i == 0) {
+        $self->{output}->add_option_msg(short_msg => 'No monitors where checked, please check filter_caption parameter and api response.');
+        $self->{output}->option_exit();
+    }
+}
+
+sub custom_status_output {
+    my ($self, %options) = @_;
+    my $res = sprintf(
+        "'%s' status : '%s', message is '%s'",
+        $self->{result_values}->{extendedcaption},
+        $self->{result_values}->{state},
+        $self->{result_values}->{messagetext}
+    );
+    return $res;
+}
+# as --filter-caption allow to filter element to check and this api don't allow parameter filtering, we should cache
+# the output in case a client make multiples check a minute.
+sub request_monitors {
```

Comment:
Again you split manage_selection function in two but I'm not sure to understand why